src/librustc_parse/parser/diagnostics.rs

```diff
@@ -934,6 +934,25 @@ impl<'a> Parser<'a> {
             return self.expect(&token::Semi).map(drop);
         } else if !sm.is_multiline(self.prev_token.span.until(self.token.span)) {
             // The current token is in the same line as the prior token, not recoverable.
+        } else if sm.is_multiline(self.prev_token.span.until(self.token.span))
+            && [token::Comma, token::Colon].contains(&self.token.kind)
+            && &self.prev_token.kind == &token::CloseDelim(token::Paren)
+        {
+            // Likely typo: The current token is on a new line and is expected to be
+            // `.`, `;`, `?`, or an operator after a close delimiter token.
+            //
+            // let a = std::process::Command::new("echo")
+            //         .arg("1")
+            //         ,arg("2")
+            //         ^
```

Comment:
I think we could _potentially_ add a case for this specific scenario to detect typos (like `;`, `:` and `,`), but I would like to see what the error looks like if `self.expect(&Semi)` where called instead to see how misleading it would be.

Reply:
with this change to my original block:

```rust
else if [token::Comma, token::Colon].contains(&self.token.kind)
        && &self.prev_token.kind == &token::CloseDelim(token::Paren)
        
    self.expect(&token::Semi)?;
}
```

I get this error message:

```
error: expected one of `.`, `;`, `?`, or an operator, found `,`
 --> src/main.rs:4:5
  |
3 |     .arg("1")
  |              - expected one of `.`, `;`, `?`, or an operator
4 |     ,arg("2")
  |     ^ unexpected token

error: aborting due to previous error

error: could not compile `tester`.
```

That seems pretty close.  It would be nice to drop the "expected one of..." line down to the line 4 level below the unexpected token message IMO.